src/Staking.sol

```diff
@@ -265,8 +265,11 @@ contract Staking is ERC721Holder, Ownable {
                 j = unsafe_inc(j)
             ) {
                 uint256 tokenId = _deposits[account][bgContract].at(j);
-                amount += (getRewardsForToken(account, tokenId, i));
-                depositBlocks[bgContract][tokenId] = block.number;
+                uint256 thisAmount = (getRewardsForToken(account, tokenId, i));
+                if (thisAmount > 0) {
+                    amount += thisAmount;
+                    depositBlocks[bgContract][tokenId] = block.number;
```

Comment:
previously, `depositBlocks[bgContract][tokenId] = block.number;` was set even if the rewards were 0, meaning the day's progress towards a reward would be lost